ingestion/src/metadata/ingestion/ometa/mixins/role_policy_mixin.py

```diff
@@ -0,0 +1,440 @@
+
+    To be inherited by OpenMetadata
+    """
+
+    client: REST
+
+    def _validate_instance_description(
+        self, entity: Type[T], entity_id: Union[str, basic.Uuid]
+    ) -> Optional[T]:
+        """
+        Validates if we can update a description or not. Will return
+        the instance if it can be updated. None otherwise.
+
+        Args
+            entity (T): Entity Type
+            entity_id: ID
+            description: new description to add
+            force: if True, we will patch any existing description. Otherwise, we will maintain
+                the existing data.
+        Returns
+            instance to update
+        """
+
+        instance = self.get_by_id(entity=entity, entity_id=entity_id, fields=["*"])
+
+        if not instance:
+            logger.warning(
+                f"Cannot find an instance of '{entity.__class__.__name__}' with id [{str(entity_id)}]."
+            )
+            return None
+
+        return instance
+
+    @staticmethod
+    def _get_rule_merge_patches(
+        previous: List,
+        current: List,
+        rule_index: int,
+        path: str,
+        is_enum: bool,
+    ) -> List[Dict]:
+        """
+        Get the operations required to overwrite the set (resources or operations) of a rule.
+        :param previous: the previous set to be overwritten by current
+        :param current: the current set to overwrite previous
+        :param rule_index: the index of the rule on which we are being operated
+        :param path: the formattable string that names the path
+        :param is_enum: is the set enums or not
+        :return: List of patch operations
+        """
+        data: List[Dict] = []
+        for index in range(len(previous) - 1, len(current) - 1, -1):
+            data.append(
+                {
+                    OPERATION: REMOVE,
+                    PATH: path.format(rule_index=rule_index - 1, index=index),
+                }
+            )
+        index: int = 0
+        for item in current:
+            data.append(
+                {
+                    OPERATION: REPLACE if index < len(previous) else ADD,
+                    PATH: path.format(rule_index=rule_index - 1, index=index),
+                    VALUE: item.name if is_enum else item,
+                }
+            )
+            index += 1
+        return data
+
+    @staticmethod
+    def _get_optional_rule_patch(
+        previous: Union[basic.FullyQualifiedEntityName, basic.Markdown],
+        current: Union[basic.FullyQualifiedEntityName, basic.Markdown],
+        rule_index: int,
+        path: str,
+    ) -> List[Dict]:
+        """
+        Get the operations required to update an optional rule field
+        :param previous: the field from the previous rule
+        :param current: the field from the current rule
+        :param rule_index: the index of the previous rule
+        :param path: path string for the filed
```

Comment:
nit - Sometimes we use `:param ....:` and sometimes `Args...` for the docstring. Could we keep it all as `Args` please?